modules/raft/src/main/java/org/apache/ignite/internal/raft/RaftGroupServiceImpl.java

```diff
@@ -449,11 +449,20 @@ public <R> CompletableFuture<R> run(Command cmd) {
             return refreshLeader().thenCompose(res -> run(cmd));
         }
 
-        Function<Peer, ActionRequest> requestFactory = targetPeer -> factory.actionRequest()
-                .command(cmd)
-                .groupId(groupId)
-                .readOnlySafe(true)
-                .build();
+        Function<Peer, ActionRequest> requestFactory;
```

Comment:
Suggestion: How about creating a separate, understandable functional interface with a description? Even with implementations to make it clearer why this is.

Reply:
Not here, it's not relevant to my changes